File: src/main/java/graph/edge/DuplexChannelFactory.java
package graph.edge;

import graph.vertex.Vertex;
import org.jetbrains.annotations.NotNull;

public class DuplexChannelFactory implements ChannelFactory {
    @Override
    public Edge createChannel(@NotNull Vertex from, @NotNull Vertex to, int weight, double errors) {
        return new DuplexChannel(from, to, weight, errors);
    }

    @Override
    public String toString() {
        return "Duplex";
    }
}

File: src/main/java/mvc/model/channel/DuplexChannelFactory.kt
package mvc.model.channel

import mvc.model.node.Node

class DuplexChannelFactory : ChannelFactory {
    override fun createChannel(from: Node, to: Node, weight: Int, errors: Double): Channel {
        return DuplexChannel(from, to, weight, errors)
    }

    override fun toString(): String {
        return "Дуплексний канал"
    }
}
